python-package/lightgbm/compat.py

```diff
@@ -140,6 +148,8 @@ class _LGBMRegressorBase:  # type: ignore
     _LGBMCheckClassificationTargets = None
     _LGBMComputeSampleWeight = None
     _LGBMValidateData = None
+    _sklearn_ClassifierTags = None
+    _sklearn_RegressorTags = None
```

Comment:
Because if an earlier `ImportError` happens, then this part above would never get evaluated:

```python
    except ImportError:
        _sklearn_ClassifierTags = None
        _sklearn_RegressorTags = None
```

And then the `from .compat imprt _sklearn_ClassifierTags` in `sklearn.py` would fail. This is not new, just following the pattern that's existed in LightGBM for a long time (see all the other `{something} = None` above it).

Happy to consider something else if you have a recommendation for improving this!

Reply:
Oh yeah, sorry.

The first one is defined for scikit-learn<1.6 and the second when scikit-learn isn't installed.

Reply:
yeah exactly. No need to be sorry, it's confusing!